frontend/tasks/scripts.js

```diff
@@ -55,36 +61,72 @@ gulp.task('scripts-misc', () => {
 });
 
 gulp.task('scripts-generate-static-html', () => {
-  return commonBrowserify('generate-static-html.js', browserify({
+  return browserify({
     entries: [config.SRC_PATH + 'generate-static-html.js'],
     debug: config.IS_DEBUG,
     fullPaths: config.IS_DEBUG,
     transform: [babelify],
-    standalone: 'generate-static-html'
-  }).external(vendorModules));
+    standalone: 'generate-static-html',
+    bundleExternal: false
+  }).external(vendorModules)
+    .bundle()
+    .pipe(source('generate-static-html.js'))
+    .pipe(buffer())
+    .pipe(sourcemaps.init({loadMaps: true}))
+     // don't uglify in development. eases build chain debugging
+    .pipe(gulpif(!config.IS_DEBUG, uglify()))
+    .on('error', gutil.log)
+    .pipe(sourcemaps.write('./'))
+    .pipe(gulp.dest(config.DEST_PATH));
 });
 
-gulp.task('scripts-app-main', () => {
+gulp.task('scripts-app', () => {
   return commonBrowserify('app.js', browserify({
-    entries: [config.SRC_PATH + 'app/index.js'],
+    entries: [config.SRC_PATH + 'app.js'],
     debug: config.IS_DEBUG,
     fullPaths: config.IS_DEBUG,
-    transform: [babelify]
+    transform: [babelify],
+    standalone: 'app',
+    bundleExternal: false
   }).external(vendorModules));
 });
 
+gulp.task('scripts-app-individual', () => {
```

Comment:
It actually still is, because we use these scripts to do the server-side rendering. Not ideal, but maybe I can revisit this once I have more experience with the code in practice.

I did try to split things out so that all the app code was in a bundle and each of the scripts-app-individual bundles were very very tiny, but I had trouble getting the browserify external stuff to work the way I wanted. I think every single subpackage needs to be specified as being external, and not just a top level package? Hopefully I'm wrong about that.